Update macOS version detection through 13 Ventura

dbsoft opened the pull request:
This updates version detection and the blacklist. It also allows webrtc to find SDKs 11.0 and higher.  Also standardize the minimum version of MacOS as 10.7.  There were 3 different versions: mozconfig specified 10.7, Info.plist 10.5 and the version detection code 10.6.  Can revisit this if someone wants to build for older versions, but I only have Intel machines 10.7 and higher.  Also force light mode on newer SDKs via Info.plist.

These changes lay the groundwork for Apple Silicon builds if anyone is interested in Waterfox Classic on Apple Silicon.

dbsoft committed:
Issue #94 - Part 1 - Update macOS version detection through 13 Ventura

This updates version detection and the blacklist. It also allows webrtc to find SDKs 11.0 and higher.  Also standardize the minimum version of MacOS as 10.7.  There were 3 different versions: mozconfig specified 10.7, Info.plist 10.5 and the version detection code 10.6.  Can revisit this if someone wants to build for older versions, but I only have Intel machines 10.7 and higher.  Also force light mode on newer SDKs via Info.plist.

These changes lay the groundwork for Apple Silicon builds if anyone is interested in Waterfox Classic on Apple Silicon.

dbsoft commented:
Is Waterfox Classic inactive or on the back burner or something? There have been 125 commits in the main Waterfox repository since I made this pull request. Not a peep on the 2 active pull requests or the related issue.

Felipefpl commented:
Indeed, it's not possible at least a few of these commits couldnt come to us.

adamp01 commented:
Thanks for this @dbsoft - we have been focused on getting G5 shipped but I will look over this and get it merged sometime this week.

dbsoft commented:
Might have another commit coming, now that Ventura has been released I should be able to test on it. I had to do a fix for White Star, might need one for Waterfox too... trying to get Ventura installed on an unsupported Intel Mac today.  So hold off on merging until I update this comment. Thanks!

Ok still working on the fix, the one I used in White Star fixed everything, but in Waterfox it fixes the main windows but not popup windows, I can't really see why.

adamp01 commented:
Really appreciate you working to solve this @dbsoft - I'll leave this open until you say otherwise.

dbsoft commented:
I had to stop looking into the popup window problems, until this weekend due to work and family obligations, but I have this as top priority to get a fix out after the new White Star release on Tuesday.  

Another minor issue that is slowing down my progress, Waterfox XUL takes like 30 minutes to link on my 2013 MacBook Pro... I know it is an older machine but White Star XUL only takes a few minutes.  Anyone know why ld64 takes so long on Waterfox XUL and is there a way to speed it up?

dbsoft committed:
Issue #94 and #164 - Part 2 - Fix transparent windows on Ventura

This fix was in two parts... first the same fix as White Star by ensuring contentView is not nil.  A second fix was required based on Mozilla bug https://bugzilla.mozilla.org/show_bug.cgi?id=1576387 Move the setContentView override from BaseWindow to ToolbarWindow.  This stops the setContentView override from screwing up the PopupWindows.

dbsoft commented:
Ugh that was supposed to be Issue #165.... but can people test this fix and see if it works without causing other issues?

If you can't compile your own and want to test, you can download the build I am using here: https://dbsoft.org/whitestar/waterfox-classic-2022.08.en-US.mac.dmg

dbsoft commented:
@MrAlex94 Thanks for merging! The fix is pretty critical with Ventura being released... it works for me but I haven't gotten feedback yet so I hope it doesn't cause any other problems!
